### DISCUSSION TITLE
Search bookmark by looking in archived content

### DISCUSSION THREAD

**Discussion by MageSlayer** [go-shiori/shiori#715] [Feature requests]
Hi

Perhaps I am doing something wrong, but it looks like shiori currently does not look in archived content when searching. e.g. my link archive contains some word, but searching for it does not show any results.

Has shiori this functionality already or it should be a feature request instead?

Thanks

**Comment by fmartingr** [comment]
We don't currently search in the archived content since thats a file stored in the filesystem and searches would be very expensinve. We use the readable content though, so if they keyword you are looking for is there the bookmark should show up.

**Comment by MageSlayer** [comment]
Not sure, what "readable content" is, but searching within page content does not work.

**Reply by fmartingr** [reply]
When you store a bookmark, we store by default a readable or article version of it by default, and optionally if checked, the archived version (which should be a copy of the site as you see in on the browser in most cases). This article version is used in search while the archive does not.

You can check if your bookmark has a readable version if it has a icon like this on the title:  ![Screenshot 2023-09-09 at 09 25 44](https://github.com/go-shiori/shiori/assets/812088/1254faa3-6b9a-4d88-b350-9c3435146f9e)

Readable version looks like this:
![Screenshot 2023-09-09 at 09 26 02](https://github.com/go-shiori/shiori/assets/812088/cc81ab36-49c9-4133-861b-296a945e768c)

**Comment by MageSlayer** [comment]
Strange, but it looks like that search is unreliable.
For example, I've used "https://thehealthylife.com/" as my bookmark. It contains icon you've mentioned.

Now, I can clearly see that "readable content" contains word "Markus" at the end.
However trying to search for it, Shiori says "No saved bookmarks yet :("

**Reply by fmartingr** [reply]
That may be a bug then. Can you fill an issue specifying your version and database engine so we can take a look?

**Reply by MageSlayer** [reply]
https://github.com/go-shiori/shiori/issues/722
